### PR TITLE
Update Gradle, AGP and Dokka + Java version bump for the plugin module

### PR DESCRIPTION
**Pull request by kristinakoneva**
## Summary

Bump Android Gradle Plugin, Dokka, and the Gradle wrapper so the Dependabot autodismiss workflow can treat the related build-tooling alerts as **trusted + current** and dismiss them.

**Important:** this does **not** actually fix the vulnerabilities. Even on the latest AGP (**9.3.0**) and Dokka (**2.2.0**), those plugins still pull in the same vulnerable transitive versions (`bcprov-jdk18on` **1.79**, `jackson-databind` **2.15.3**). There is no upstream release yet that ships patched Jackson/Bouncy Castle.

What this change *does* do is satisfy the dismisser’s **trusted + current** rule. Once AGP and Dokka are on the latest versions, the workflow can classify these alerts as build-tooling / trusted-source transitives (not production-shipping) and auto-dismiss them. The CVEs remain on the build classpath until Google/JetBrains bump their own deps.

These alerts are for transitive dependencies of AGP and Dokka only. I assume they are not shipped in Halley’s published library artifacts and are not runtime dependencies of consumers if they are eligible to be dismissed by the workflow. Let me know if there is something I missed.

### Dependency updates
| Dependency | Before | After |
| --- | --- | --- |
| Android Gradle Plugin | 9.1.0 | **9.3.0** |
| Dokka | 2.1.0 | **2.2.0** |
| Gradle wrapper | 9.3.1 | **9.5.0** (required by AGP 9.3) |

### Related change
- Raise `halley-plugin` JVM target from **11 → 17**. AGP 9.3 publishes Java 17 variants only; keeping the plugin on 11 breaks Dokka / `compileClasspath` resolution (`No matching variant` for `com.android.tools.build:gradle:9.3.0`). Library modules remain on Java 8 for consumer compatibility.
### Dependabot alerts (expected dismissals — not a security patch)
After merge, run (or wait for) the **Dismiss Dependabot alerts for dependencies not used in production** workflow with `dry-run=false`. With AGP and Dokka on latest, the currency check should pass and these open alerts should be auto-dismissed as build-tooling / trusted-source transitives:
- https://github.com/infinum/android-halley/security/dependabot/97 — Critical: `org.bouncycastle:bcprov-jdk18on` (GOST CTR keystream reuse; via AGP)
- https://github.com/infinum/android-halley/security/dependabot/94 — High: `jackson-databind` PTV array subtype allowlist bypass (via Dokka)
- https://github.com/infinum/android-halley/security/dependabot/93 — High: `jackson-databind` PTV generic type-parameter bypass (via Dokka)
- https://github.com/infinum/android-halley/security/dependabot/96 — Moderate: `jackson-databind` case-insensitive `@JsonIgnoreProperties` bypass (via Dokka)
- https://github.com/infinum/android-halley/security/dependabot/95 — Moderate: `jackson-databind` `InetSocketAddress` eager DNS / SSRF (via Dokka)
> Note: this does **not** patch Jackson/Bouncy Castle inside AGP/Dokka themselves. It unblocks our policy-based dismissal for non-production / buildscript usage while we stay on current trusted top-level plugins.

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [X] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

I ran the sample app & deployed a local version of the lib successfully.

### CODE REVIEW

File: gradlew.bat [hunk 1]
```diff
@@ -51,7 +51,7 @@ echo. 1>&2
 echo Please set the JAVA_HOME variable in your environment to match the 1>&2
 echo location of your Java installation. 1>&2
 
-goto fail
+"%COMSPEC%" /c exit 1
```
[the remark this answers is not in the document]
Comment (kristinakoneva) [reply gradlew.bat:54]: I didn't update this manually, I ran `gradlew wrapper --gradle-version=9.5.0 ` to update the wrapper files.

File: gradlew.bat [hunk 2]
```diff
@@ -65,30 +65,18 @@ echo. 1>&2
...
```
[the remark this answers is not in the document]
Comment (kristinakoneva) [reply gradlew.bat:68]: I didn't update this manually, I ran `gradlew wrapper --gradle-version=9.5.0 ` to update the wrapper files.

File: gradlew.bat [hunk 3]
```diff
@@ -65,30 +65,18 @@ echo. 1>&2
...
 @rem Setup the command line
 
-set CLASSPATH=%APP_HOME%\gradle\wrapper\gradle-wrapper.jar
 
 
 @rem Execute Gradle
-"%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -classpath "%CLASSPATH%" org.gradle.wrapper.GradleWrapperMain %*
+@rem endlocal doesn't take effect until after the line is parsed and variables are expanded
+@rem which allows us to clear the local environment before executing the java command
+endlocal & "%JAVA_EXE%" %DEFAULT_JVM_OPTS% %JAVA_OPTS% %GRADLE_OPTS% "-Dorg.gradle.appname=%APP_BASE_NAME%" -jar "%APP_HOME%\gradle\wrapper\gradle-wrapper.jar" %* & call :exitWithErrorLevel
```
[the remark this answers is not in the document]
Comment (kristinakoneva) [reply gradlew.bat:78]: I didn't update this manually, I ran `gradlew wrapper --gradle-version=9.5.0 ` to update the wrapper files.

File: halley-plugin/build.gradle [hunk 4]
```diff
@@ -8,21 +8,21 @@ kotlin {
 }
 
 java {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
+    sourceCompatibility JavaVersion.VERSION_17
+    targetCompatibility JavaVersion.VERSION_17
 }
```
[the remark this answers is not in the document]
Comment (kristinakoneva) [reply halley-plugin/build.gradle:13]: This should be documented once we release a new version of the library.

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -1,14 +1,14 @@
 [versions]
 halley = "1.0.0"
-gradle = "9.1.0"
+gradle = "9.3.0"
```
[the remark this answers is not in the document]
Comment (kristinakoneva) [reply gradle/libs.versions.toml:4]: Added comment.

**Review by KCeh — APPROVED** [review]
Looks good 🚀

File: halley-plugin/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (KCeh) [reply halley-plugin/build.gradle:13]: Let's just keep this in mind, it sounds like a breaking change = major version bump
